File: src/RealWorld.Api/Models/Article.cs
using System.Text.RegularExpressions;

namespace RealWorld.Api.Models;

public class Article
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string UserId { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public User? Author { get; set; }
    public List<ArticleTag> ArticleTags { get; set; } = new();
    public List<ArticleFavorite> Favorites { get; set; } = new();
    public List<Comment> Comments { get; set; } = new();

    public void Update(string? title, string? description, string? body)
    {
        if (!string.IsNullOrEmpty(title))
        {
            Title = title;
            Slug = ToSlug(title);
            UpdatedAt = DateTime.UtcNow;
        }
        if (!string.IsNullOrEmpty(description))
        {
            Description = description;
            UpdatedAt = DateTime.UtcNow;
        }
        if (!string.IsNullOrEmpty(body))
        {
            Body = body;
            UpdatedAt = DateTime.UtcNow;
        }
    }

    public static string ToSlug(string title)
    {
        return Regex.Replace(title.ToLowerInvariant(), @"[\&|\uFE30-\uFFA0|\'|\""\s\?\,\.]+", "-");
    }
}

File: comment-service/src/main/java/io/spring/core/article/Article.java
package io.spring.core.article;

import java.util.UUID;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.joda.time.DateTime;

@Getter
@NoArgsConstructor
@EqualsAndHashCode(of = {"id"})
public class Article {
  private String userId;
  private String id;
  private String slug;
  private String title;
  private String description;
  private String body;
  private DateTime createdAt;
  private DateTime updatedAt;

  public Article(String title, String description, String body, String userId) {
    this.id = UUID.randomUUID().toString();
    this.slug = toSlug(title);
    this.title = title;
    this.description = description;
    this.body = body;
    this.userId = userId;
    this.createdAt = new DateTime();
    this.updatedAt = new DateTime();
  }

  private static String toSlug(String title) {
    return title.toLowerCase().replaceAll("[\\s\\?\\,\\.]+", "-");
  }
}
